### ISSUE TITLE
Export Active Project should not export "empty" stuff

### ISSUE THREAD

**Issue by Mo-Gul** [rubberduck-vba/Rubberduck#5834]
**Justification**
Currently also modules and sheets are exported that only contain (stripped) "Option Explicit".

**Description**
It would be nice if there would be an option to suppress the export of modules/sheets that only contain the whitespace and `Option Explicit`.

**Additional context**
`Option Explicit` is added automatically to every module/sheet/class/form when it is opened the first time in VBE and the VBE option `Require Variable Declaration` is enabled. So when someone not uses Rubberduck (don't know why people do not ;)) and wants to find out where is code in an Excel file, one has to look at every entry in the Project Explorer. And if there was no code, latest now there is ...

Of course one could delete all these `Option Explicit`s, but if you have 50 sheets ... I think it is much more convenient to declare a sheet/module/class/form empty, if it only contains whitespace and `Option Explicit`.

**Comment by Mo-Gul** [comment]
@retailcoder, could you give me a hint where I can find the relevant stuff in the code here as well. Maybe then I'll be able to provide a PR too.

**Comment by retailcoder** [comment]
Hi @Mo-Gul, you'll find that command under Core.UI.Commands.ComCommands; the call that exports a project is around line 122:

https://github.com/rubberduck-vba/Rubberduck/blob/1e819c8e2602da623a5c9b5d3734f0696b59a754/Rubberduck.Core/UI/Command/ComCommands/ExportAllCommand.cs#L120-L125

It calls into the COM wrapper for `VBProject`, which has an implementation for VBA, and another one for VB6:

https://github.com/rubberduck-vba/Rubberduck/blob/1e819c8e2602da623a5c9b5d3734f0696b59a754/Rubberduck.VBEditor.VBA/SafeComWrappers/VB/VBProject.cs#L156-L166

I think that's where a check should be added to skip exporting empty modules. If we want to prompt and make it optional, the signature here needs to change to take a `bool exportEmptyComponents` parameter.

**Comment by MDoerner** [comment]
I think this should definitely be optional. For document modules, it does not really make a difference, since we cannot recreate them anyway, but for other modules not exporting them would mean that the empty modules would vanish on sync. That would be a breaking change.

We actually already have another place where we determine whether a module is empty, the [EmptyModuleInspection](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.CodeAnalysis/Inspections/Concrete/EmptyModuleInspection). That might give you an idea how we currently define being empty.

**Comment by BZngr** [comment]
I'm wondering if adding a DeleteModuleQuickFix option for the `EmptyModuleInspection` might be a better way to help a user clean up empty modules rather than during the export process.

**Comment by retailcoder** [comment]
@BZngr why not both? ☺️ I think a `RemoveEmptyModuleQuickFix` is definitely something we should be offering when facing an `EmptyModuleInspection` result, _and_ there should be a setting to control whether or not empty modules should be exported (no need for a prompt then).

**Comment by MDoerner** [comment]
Actually, the quickfix would not solve the main problem that we export semantically empty document modules. We definitely cannot remove the from the VBA project, but they just clutter the export.

**Comment by Mo-Gul** [comment]
I agree to @retailcoder and @MDoerner. The inspection (and a possible fix) is independent from the export of an empty "component" where a component can also be a worksheet. While the code part of the worksheet can be "empty" the other parts of the worksheet _can_ have data and thus it shouldn't be removed by a fix.

I think the best option, if "empty" components should be exported or not, would be to add a checkbox somewhere in the settings. Then the user can decide what he/she prefers.

**Comment by Mo-Gul** [comment]
Because I don't have clue about C# unfortunately I don't understand _anything_ what is coded in [EmptyModuleInspection.cs](https://github.com/rubberduck-vba/Rubberduck/blob/next/Rubberduck.CodeAnalysis/Inspections/Concrete/EmptyModuleInspection.cs).

So the only thing I came up with is, is to add something like

```csharp
// BUG: not sure how to get the text of the component. The `text` part is a guess.
// TODO: Remove the "invisible header" (for components other than DOCCLS)
string componentText = component.text.Trim();
if (string.Equals(componentText, "Option Explicit", StringComparison.OrdinalIgnoreCase))
{
    continue;
}
else if (componentText.Length == 0)
{
    continue;
}
```

before line 164 in

https://github.com/rubberduck-vba/Rubberduck/blob/1e819c8e2602da623a5c9b5d3734f0696b59a754/Rubberduck.VBEditor.VBA/SafeComWrappers/VB/VBProject.cs#L156-L166

As you can see from the comments in the code, there is still some work left and most likely my suggested code is far from being optimal (if it works at all).

So unfortunately I have to give up to solve this feature request myself ... :sob:
